File: 1.StrategyPattren/C++/AccountStrategyPattern/AccountStrategyPattern/AccountSimulation.cpp
#include "Account.h"
#include "NormalAccount.h"
#include "MinusAccount.h"
#include "SavingAccount.h"

int main(){

	Account *normalAccount = new NormalAccount(50000);
	normalAccount->toString();
	normalAccount->withdraw(40000);
	normalAccount->toString();
	cout << endl;
	
	Account *minusAccount = new MinusAccount(50000, 100000);
	minusAccount->toString();
	minusAccount->withdraw(40000);
	minusAccount->toString();
	minusAccount->withdraw(150000);
	minusAccount->toString();
	cout << endl;
	
	Account *savingAccount = new SavingAccount(50000, 2015, 10, 10);
	savingAccount->toString();
	savingAccount->withdraw(40000);
	savingAccount->toString();

	return 0;
}

File: 1.StrategyPattren/JAVA/src/account/AccountSimulation.java
package account;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class AccountSimulation {
	
	@SuppressWarnings("deprecation")
	public static void main(String[] args) throws ParseException{
		Account normalAccount = new NomalAccount(50000);
		System.out.println(normalAccount);
		normalAccount.withdraw(40000);
		System.out.println(normalAccount);
		System.out.println();
		
		Account minusAccount = new MinusAccount(50000, 100000);
		System.out.println(minusAccount);
		minusAccount.withdraw(40000);
		System.out.println(minusAccount);
		minusAccount.withdraw(150000);
		System.out.println(minusAccount);
		System.out.println();
		
		SimpleDateFormat format = new SimpleDateFormat( "yyyy-MM-dd" );
		Date finishDate = format.parse("2015-10-10");
		Account savingAccount = new SavingAccount(50000, finishDate);
		System.out.println(savingAccount);
		savingAccount.withdraw(40000);
		System.out.println(savingAccount);
	}
}
